lib/presenter/Chart.js

```diff
@@ -552,28 +538,28 @@ class Chart extends Events {
             return;
         }
 
-            // Wait for a mouse click before enabling zooming
-        opts.d3Svg.on('mouseup.enable_zooming', () => {
+        // Wait for a mouse click before enabling zooming
+        this.d3Svg.on('mouseup.enable_zooming', () => {
             this._enableZooming(opts);
-            opts.d3Svg.on('mouseup.enable_zooming', null);
+            this.d3Svg.on('mouseup.enable_zooming', null);
```

Comment:
The majority of the changes that follow are just a result of referencing member vars rather than the monster opts object